revenue/revenue.go

```diff
@@ -56,91 +47,68 @@ func GetRevenueReport(cfg *Config) (*Report, error) {
-// outpoint strings.
-func getEvents(channelIDs map[lnwire.ShortChannelID]string, queryForwards func(
-	offset, maxEvents uint32) (*lndclient.ForwardingHistoryResponse,
-	error)) ([]revenueEvent, error) {
+// getEvents produces a set of revenue events from a set of forwarding events
+// that can be used to create a report.
+func getEvents(channelIDs map[lnwire.ShortChannelID]string,
+	fwdEvents []lndclient.ForwardingEvent) []revenueEvent {
 
+	// Do not prealloc because we may not add events if we cannot get their
+	// channel ID from the channelIDs map.
+	// nolint: prealloc
 	var events []revenueEvent
 
-	// Query is a helper function which adds forwarding events to our set
-	// of events, and returns the new offset and number of events returned.
-	// We use this function with our generic paginater to build up a set of
-	// forwarding events from the paginated api.
-	query := func(offset, maxEvents uint64) (uint64, uint64, error) {
-		resp, err := queryForwards(
-			uint32(offset), uint32(maxQueryEvents),
+	// Get the event's channel outpoints from out known list of maps and
+	// create a revenue event. Return an error if the short channel id's
+	// outpoint cannot be found, because we expect all known short channel
+	// ids to be provided.
+	for _, fwd := range fwdEvents {
+		shortChanIn := lnwire.NewShortChanIDFromInt(
+			fwd.ChannelIn,
 		)
-		if err != nil {
-			return 0, 0, err
-		}
+		shortChanOut := lnwire.NewShortChanIDFromInt(
+			fwd.ChannelOut,
+		)
+
+		incoming, ok := channelIDs[shortChanIn]
+		if !ok {
+			log.Errorf("cannot find channel "+
+				"incoming outpoint for forward: %v(%v "+
+				"msat) -> %v(%v msat)", shortChanIn,
+				fwd.AmountMsatIn, shortChanOut,
+				fwd.AmountMsatOut)
 
-		// Get the event's channel outpoints from out known list of maps
-		// and create a revenue event. Return an error if the short
-		// channel id's outpoint cannot be found, because we expect all
-		// known short channel ids to be provided.
-		for _, fwd := range resp.Events {
-			shortChanIn := lnwire.NewShortChanIDFromInt(
-				fwd.ChannelIn,
-			)
-			shortChanOut := lnwire.NewShortChanIDFromInt(
-				fwd.ChannelOut,
-			)
-
-			incoming, ok := channelIDs[shortChanIn]
-			if !ok {
-				log.Errorf("cannot find channel "+
-					"incoming outpoint for forward: %v(%v "+
-					"msat) -> %v(%v msat)", shortChanIn,
-					fwd.AmountMsatIn, shortChanOut,
-					fwd.AmountMsatOut)
-
-				continue
-			}
-
-			outgoing, ok := channelIDs[shortChanOut]
-			if !ok {
-				log.Errorf("cannot find channel "+
-					"outgoing outpoint for forward: %v(%v "+
-					"msat) -> %v(%v msat)", shortChanIn,
-					fwd.AmountMsatIn, shortChanOut,
-					fwd.AmountMsatOut)
-
-				continue
-			}
-
-			events = append(events, revenueEvent{
-				incomingChannel: incoming,
-				outgoingChannel: outgoing,
-				incomingAmt:     fwd.AmountMsatIn,
-				outgoingAmt:     fwd.AmountMsatOut,
-			})
+			continue
 		}
 
-		return offset, uint64(len(resp.Events)), nil
```

Comment:
This was the problem with this function, returning offset rather than resp.LastIndexOffset.